src/material/core/focus-indicators/_focus-indicators.scss

```diff
@@ -22,12 +22,16 @@
   $border-radius: map-get($config, border-radius);
 
   // Base styles for focus indicators.
-  .mat-focus-indicator::before {
-    @include mat-fill();
-    box-sizing: border-box;
-    pointer-events: none;
-    border: $border-width $border-style transparent;
-    border-radius: $border-radius;
+  .mat-focus-indicator {
+    position: relative;
```

Comment:
This style has specificity 0,1,0, and is included before any component styles, and so technically if this mixin isn't wrapped in an additional selector, this style won't override any component's `position` style. However, in our dev app, we write:

```
.demo-strong-focus {
  @include mat-strong-focus-indicators();
}
```

causing this style to have 0,2,0 specificity. Consequently, I bumped component `position` specificity to 0,2,0. This seemed to match what we are currently doing when overriding ripple positioning.

Reply:
Alternatively, we can include this style outside of the `mat-strong-focus-indicators` mixin directly within `mat-core`.

Pros:

1. We're not adding `position: relative` to a bunch of components when this mixin is included (toggling `position` can lead to subtle bugs that may be missed).
2. It'll likely be applied with lower specificity (it's more likely that `mat-strong-focus-indicators` may be included within an activator class), meaning again it's less likely that this style accidentally overrides a component style.

Cons:

1. We'll need to run TGP (running as we speak).
2. We include a small amount of focus indicator related CSS regardless of whether the `mat-strong-focus-indicator` mixin is included.

Reply:
Resolving, we're adding to `mat-core`.